google-cloud-storage/src/main/java/com/google/cloud/storage/Option.java

```diff
@@ -29,4 +34,25 @@
   Option(O opt) {
     super(opt);
   }
+
+  @SafeVarargs
+  static <O extends Option<?>> O[] dedupe(IntFunction<O[]> gen, O... os) {
+    return dedupe(gen, Arrays.stream(os));
+  }
+
+  @SafeVarargs
+  static <O extends Option<?>> O[] dedupe(IntFunction<O[]> gen, Collection<O> collection, O... os) {
+    return dedupe(gen, Stream.of(collection.stream(), Arrays.stream(os)).flatMap(s -> s));
+  }
+
+  @SafeVarargs
+  static <O extends Option<?>> O[] dedupe(IntFunction<O[]> gen, O[] array, O... os) {
+    return dedupe(gen, Stream.of(Arrays.stream(array), Arrays.stream(os)).flatMap(s -> s));
+  }
+
+  private static <O extends Option<?>> O[] dedupe(IntFunction<O[]> gen, Stream<O> s) {
```

Comment:
IIUC this method will 1) group by option class type 2) get last instance from group then 3) return array of this filtering.

Please add a comment describing what each of these helpers does. Change LGTM.

Reply:
All `Option`s contain an `Opt`, `Opt`s are distinct classes allowing us to group based on those classes. Once grouped, we select the last element to provide last wins.

Each of these helpers is an internal implementation detail, primarily due to the fact that generic arrays can not be instantiated in Java and requires a factory to be passed in.

Reply:
Thanks Ben, please add this context as comments in the code for future us.